autorest/codegen/__init__.py

```diff
@@ -134,7 +149,10 @@ def _create_code_model(self, yaml_data: Dict[str, Any], options: Dict[str, Union
         code_model.setup_client_input_parameters(yaml_data)
 
         # Get my namespace
-        namespace = self._autorestapi.get_value("namespace")
+        if not code_model.options.get("package_mode"):
```

Comment:
i don't think we should change this code, we really want package mode to be as similar to generation flags as possible, I don't think it helps the users to have these slightly different flags for generation and package mode